src/_pytest/main.py

```diff
@@ -592,7 +592,7 @@ def filter_(f):
                 col = self._node_cache[argpath]
             else:
                 collect_root = self._pkg_roots.get(argpath.dirname, self)
-                col = collect_root._collectfile(argpath)
+                col = collect_root._collectfile(argpath, handle_dupes=False)
```

Comment:
That's wrong.. and will not handle dupes at all then.
Also the output of the test should be matched (includes `sub/test_file.py` twice).

Reply:
Likely works better with `add_to_dupes` (https://github.com/pytest-dev/pytest/pull/4748).